### ISSUE TITLE
lifecycle.graceTimeOut is not honored for WebSocket connections

### ISSUE THREAD

**Issue by erhudy** [containous/traefik#3630]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

We are supporting an application that makes use of long-lived WebSocket connections to stream VoIP audio data. In order to give this application a reasonable amount of time to complete calls that are in progress when the Traefik DaemonSet is updated, I have set `lifecycle.graceTimeOut` to a very long period, per the configuration file below. Additionally, `terminationGracePeriodSeconds` in the DaemonSet's PodSpec has been set to the same numeric value. However, when I test this by triggering a rolling update on the daemonset, Traefik terminates within 10-20 seconds, dropping the active WebSocket collections on the floor.

### What did you expect to see?

Traefik should delay restart until all WebSockets are closed or it has reached the maximum possible timeout (`lifecycle.requestAcceptGraceTimeout` + `lifecycle.graceTimeOut`).

### What did you see instead?

Traefik terminates shortly after reaching `lifecycle.requestAcceptGraceTimeout` and does not give the existing WebSocket connections time to end gracefully. 

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.5
Codename:     tetedemoine
Go version:   go1.10.3
Built:        2018-07-10_03:54:03PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http"]
logLevel = "INFO"

[entryPoints]
  [entryPoints.http]
    address = ":80"

  [entryPoints.metrics]
    address = ":9301"

[traefikLog]

[kubernetes]

[lifecycle]
  requestAcceptGraceTimeout = "10s"
  graceTimeOut = "3600s"

[metrics]
  [metrics.prometheus]
    entryPoint = "metrics"
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
(paste your output here)
```

**Comment by traefiker** [comment]
Closed by #3636.

**Comment by traefiker** [comment]
Closed by #3777.
